src/types/union.ts

```diff
@@ -966,6 +968,46 @@ export function Union(...alternatives: Rt[]): any {
 
   return create(
     (value, visited) => {
+      if (
+        alternatives.length > 1 &&
```

Comment:
Why is this condition necessary?

Reply:
I wouldn’t call it necessary, just a performance optimization that we can easily get rid off. If you think its not worth it I can remove it for sure. It might help with coverage as well. 

Also we can implement some of the checks with type guards and that would help with coverage as well, but I figured I’d use existing types as much as possible. The current way Reflect is implemented I can’t see a way to reuse the types for typeguards.

Reply:
Yeah, I think for simplicity's sake we should remove it.